### PR TITLE
Added side navbar

### LINKED ISSUE #6: use sidebar for webui
hi,

thx for nice webui great work .

but i prefer we need to use the same method like other social  platforms like tribler, zeronet,....
top Bar is mostly used for search, menu, actions, notify count, settings or account stuff
but for page navigation is best to use a left side SideBar.
image

https://forum.tribler.org/t/tribler-web-interface/4233

### PR DESCRIPTION
**Pull request by gunjan543**
Closes #6 
sidenavbar

### CODE REVIEW

**Comment by Kumaravinash9** [comment]
Hello @gunjan543, it looks good but it requires some changes like 

* I think we should put the inbox part in the top.
improv1

* People  tab  has completely changed. So, I think you should  do `git pull` , then do changes.
improve2
 
* same problem with files tab && config tab too. It should be in the top 
improve3

**Comment by gunjan543** [comment]
inbox
It looks good in my browser

**Comment by gunjan543** [comment]
Ok thankyou @Kumaravinash9  :) 
I make the suggested changes.

**Comment by gunjan543** [comment]
@Kumaravinash9 , I was thinking since the navbar is made vertical, so I can make this navbar horizontal, what do you suggest? 
horizontal_nav

**Comment by Kumaravinash9** [comment]
> @Kumaravinash9 , I was thinking since the navbar is made vertical, so I can make this navbar horizontal, what do you suggest?
> horizontal_nav

It's look good to me. Better to ask this from @defnax and @rottencandy.

**Comment by gunjan543** [comment]
> > @Kumaravinash9 , I was thinking since the navbar is made vertical, so I can make this navbar horizontal, what do you suggest?
> > horizontal_nav
> 
> It's look good to me. Better to take the suggestion from @defnax and @rottencandy.

ok thankyou

File: webui-src/app/theme.css [hunk 1]
```diff
@@ -327,10 +335,14 @@ a.tab-menu-item {
 }
 .tab-menu-item:hover {
   background-color: #999;
+  width: -webkit-fill-available;
+}
 }
```
Comment (Kumaravinash9) [inline webui-src/app/theme.css:340]: It doesnot match with any bracket. you should remove this parenthesis .

File: webui-src/app/theme.css [hunk 2]
```diff
@@ -327,10 +335,14 @@ a.tab-menu-item {
...
 .tab-menu-item.selected-tab-item {
   background-color: #3ba4d7;
   color: white;
+  width: -webkit-fill-available;
+}
 }
```
Comment (Kumaravinash9) [inline webui-src/app/theme.css:346]: Here also. remove the parenthesis.

**Review by Kumaravinash9 — CHANGES_REQUESTED** [review]

File: webui-src/app/theme.css (hunk 2 above)
Comment (gunjan543) [reply webui-src/app/theme.css]: > Here also.
Okay

**Comment by defnax** [comment]
> Hello @gunjan543, it looks good but it requires some changes like
> 
> * I think we should put the inbox part in the top.
> 
> improv1
> 
> * People  tab  has completely changed. So, I think you should  do `git pull` , then do changes.
> 
> improve2
> 
> * same problem with files tab && config tab too. It should be in the top
> 
> improve3

it must stay normaly left not top the inbox/outbox...

**Comment by defnax** [comment]
on other intranet software the webui is nice organized

https://www.exoplatform.com/blog/2013/04/18/exo-platform-4-highlights-new-navigation/
https://www.g2.com/products/exo-platform/reviews
https://imgix.lifehacker.com.au/content/uploads/sites/4/2016/03/communifire.png

File: webui-src/app/theme.css [hunk 3]
```diff
@@ -305,12 +305,20 @@ Target specific sections
...
 /* Navbar */
 nav.tab-menu {
+  border-right: 5px solid #3ba4d7;
   background-color: #333;
-  position: fixed;
-  top: 0;
-  width: 100%;
   border-bottom: 5px solid #3ba4d7;
   box-shadow: 0 5px 5px #aaa;
+  list-style: none;
```
Comment (rottencandy) [inline webui-src/app/theme.css:312]: `list-style` only affects elements which have `display:list-item` so it's not used here.

**Comment by defnax** [comment]
on qt ui

image

File: webui-src/app/theme.css [hunk 4]
```diff
@@ -305,12 +305,20 @@ Target specific sections
...
+  padding: 0;
+  margin-left: 0;
```
Comment (rottencandy) [inline webui-src/app/theme.css:314]: Padding & margin are 0 by default for `nav` I think.

File: webui-src/app/theme.css [hunk 5]
```diff
@@ -305,12 +305,20 @@ Target specific sections
...
+  float: left;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  height: 100%;
+  padding-top: 2rem;
```
Comment (rottencandy) [inline webui-src/app/theme.css:320]: Not sure if the top padding would be needed.

**Comment by defnax** [comment]
> * I think we should put the inbox part in the top.

this will never fit, we has lots of boxes, spam, attachments, starred , system, quick filters for tags and the design will not look good

File: webui-src/app/theme.css [hunk 6]
```diff
@@ -12,7 +12,7 @@ General site-wide rules
 #tab-content {
   height: 100%;
   /* Margin on top to avoid getting covered by navbar */
-  margin-top: 4em;
+  padding-top: 4em;
```
Comment (rottencandy) [inline webui-src/app/theme.css:15]: I think it's better to be more consistent with the sidebar approach, the main nav bar and the nested sidebars should be on the left.

File: webui-src/app/theme.css [hunk 7]
```diff
@@ -327,17 +335,20 @@ a.tab-menu-item {
 }
 .tab-menu-item:hover {
   background-color: #999;
+  width: -webkit-fill-available;
 }
+
 .tab-menu-item.selected-tab-item {
   background-color: #3ba4d7;
   color: white;
+  width: -webkit-fill-available;
```
Comment (rottencandy) [inline webui-src/app/theme.css:344]: Doesn't need to be specified again, previous style already defines it.

**Review by rottencandy — COMMENTED** [review]
Thanks for the changes.

I also agree that both the navbar and inner sidebar should be placed vertically on the left. It would be better this way as the navs would be consistent.

File: libwebui.so [hunk 8]
```diff
@@ -0,0 +1 @@
+libwebui.so.1.0.0
```
[the remark this answers is not in the document]
Comment (Kumaravinash9) [reply libwebui.so:1]: It would be better when we add `libwebui.so*`   in `.gitignore` file. So , in future no one will push these file .

**Comment by gunjan543** [comment]
okay, I will make the suggested changes.

**Comment by gunjan543** [comment]
I did the suggested changes, can you please review it

File: libwebui.so.1 [hunk 9]
```diff
@@ -0,0 +1 @@
+libwebui.so.1.0.0
```
Comment (rottencandy) [inline libwebui.so.1:1]: Thanks for the updates, looks like you forgot to remove the libwebui.so files, and some of the unused lines in the css.

**Comment by gunjan543** [comment]
Any other changes you wish to have?

**Comment by defnax** [comment]
> Any other changes you wish to have?

this is done or? icon on left side
image

**Comment by defnax** [comment]
i see this issues wasted space
image

image

**Comment by defnax** [comment]
it must be look more this
painted
image

**Comment by defnax** [comment]
left side navebar has issues with chats:
image

File: webui-src/app/theme.css [hunk 10]
```diff
@@ -327,17 +332,18 @@ a.tab-menu-item {
 }
 .tab-menu-item:hover {
   background-color: #999;
+  width: -webkit-fill-available;
```
Comment (rottencandy) [inline webui-src/app/theme.css:335]: So this will not work on firefox, a better solution would be to set the width along with padding & margin in `.tab-menu-item` class itself:
```diff
@@ -318,7 +316,9 @@ nav.tab-menu {
   margin-right: 0rem;
 }
 a.tab-menu-item {
-  padding: 1em;
+  padding: 1em 0;
+  margin: 0 1em;
+  width: 100%;
   text-align: center;
   text-decoration: none;
   color: #ccc;
@@ -332,18 +332,18 @@ a.tab-menu-item {
 }
 .tab-menu-item:hover {
   background-color: #999;
-  width: -webkit-fill-available;
 }
 
 .tab-menu-item.selected-tab-item {
   background-color: #3ba4d7;
   color: white;
-  width: -webkit-fill-available;
 }
```

File: webui-src/app/theme.css [hunk 11]
```diff
@@ -327,17 +332,18 @@ a.tab-menu-item {
...
   background-color: #3ba4d7;
   color: white;
+  width: -webkit-fill-available;
 }
 
 .sidebar {
-  position: fixed;
-  height: 100%;
-  width: 200px;
-  background-color: white;
```
Comment (rottencandy) [inline webui-src/app/theme.css:340]: I think the previous styles were better here, fit with the main sidebar quite well:

Comment (rottencandy) [reply webui-src/app/theme.css:340]: We don't need the `margin-top` anymore too.

File: webui-src/app/theme.css [hunk 12]
```diff
@@ -123,7 +123,9 @@ hr {
 .widget-half {
   max-width: 50%;
 }
-
+.widget-2{
+  margin-top: -12rem;
+}
```
Comment (rottencandy) [inline webui-src/app/theme.css:128]: What is this for? Don't think it's being used anywhere

File: webui-src/app/theme.css [hunk 13]
```diff
@@ -357,6 +363,9 @@ a.tab-menu-item {
   border-left: 5px solid #3ba4d7;
   animation: expand-left-border 0.1s;
 }
+.widget-2{
+  margin-top: -12rem;
+}
```
Comment (rottencandy) [inline webui-src/app/theme.css:368]: Same here

File: webui-src/app/theme.css [hunk 14]
```diff
@@ -12,7 +12,7 @@ General site-wide rules
 #tab-content {
   height: 100%;
   /* Margin on top to avoid getting covered by navbar */
-  margin-top: 4em;
```
Comment (rottencandy) [inline webui-src/app/theme.css:15]: We don't need the margin/padding at the top here either.

**Review by rottencandy — CHANGES_REQUESTED** [review]
Sorry I didn't notice this earlier, we should try to avoid using the `-webkit-*` styles since most of them will not work with non-webkit browsers(like firefox).

Also I think the chat room sidebar needs a little bit of work to fit with the navbar.

**Comment by rottencandy** [comment]
> it must be look more this
> painted
> image

This is about about mail features, so doesn't need to be done in this PR, better to handle separately.

**Comment by defnax** [comment]
> > it must be look more this
> > painted
> > image
> 
> This is about about mail features, so doesn't need to be done in this PR, better to handle separately.

yes you are  i was to fast :)

File: webui-src/app/theme.css (hunk 11 above)
Comment (gunjan543) [reply webui-src/app/theme.css:340]: > I think the previous styles were better here, fit with the main sidebar quite well:
> 
Sidenavbar looks like this in my browser, why I am getting different styling ?
Screenshot from 2021-05-06 09-15-46

File: webui-src/app/theme.css (hunk 12 above)
Comment (gunjan543) [reply webui-src/app/theme.css:128]: I have used it 11 times in app.js

File: webui-src/app/theme.css (hunk 11 above)
Comment (rottencandy) [reply webui-src/app/theme.css:340]: I had reverted your changes for the selected section here.

File: webui-src/app/theme.css (hunk 12 above)
Comment (rottencandy) [reply webui-src/app/theme.css:128]: But that is not related to the navbar changes for this PR, we can include this when we are working on app.js.

Comment (gunjan543) [reply webui-src/app/theme.css:128]: I think stylings are getting applied from there only, there are other classes also like widget-half which is providing styles to the page. I needed to move things upwards that's why I have applied that class to some elements which were not in the correct position

File: webui-src/app/theme.css (hunk 11 above)
Comment (gunjan543) [reply webui-src/app/theme.css:340]: okay

File: webui-src/app/theme.css (hunk 12 above)
Comment (rottencandy) [reply webui-src/app/theme.css:128]: Oh, so you were applying this class to elements. In that case you should commit that as well.

Comment (gunjan543) [reply webui-src/app/theme.css:128]: Do I have to commit it individually?

Comment (rottencandy) [reply webui-src/app/theme.css:128]: You can commit them in this PR itself.

Comment (gunjan543) [reply webui-src/app/theme.css:128]: ok

**Comment by gunjan543** [comment]
> left side navebar has issues with chats:
> image

I am not subscribed to any chat rooms, that's why I am not able to figure out the stylings that are getting wrong over here.
Screenshot from 2021-05-06 21-24-13
So, can you please send me some links or tell me from where I can join these chat rooms.

**Comment by defnax** [comment]
> So, can you please send me some links or tell me from where I can join these chat rooms.
https://retroshare.ch/invite

**Comment by gunjan543** [comment]
Please review, if I missed anything!

File: webui-src/app/chat/chat.css [hunk 15]
```diff
@@ -91,7 +92,7 @@
   position: absolute;
   background-color: white;
   top: 130px;
-  left: 215px;
+  left: 328px;
```
Comment (rottencandy) [inline webui-src/app/chat/chat.css:95]: Please do this to the `chatMessage` class also, So that it's consistent.

File: webui-src/app/theme.css [hunk 16]
```diff
@@ -327,17 +332,17 @@ a.tab-menu-item {
 }
 .tab-menu-item:hover {
   background-color: #999;
+ 
 }
+
 .tab-menu-item.selected-tab-item {
   background-color: #3ba4d7;
-  color: white;
```
Comment (rottencandy) [inline webui-src/app/theme.css:333]: Why is this removed? Looked better with the white highlight IMO.

File: webui-src/app/theme.css [hunk 17]
```diff
@@ -327,17 +332,17 @@ a.tab-menu-item {
...
+
 }
 
 .sidebar {
-  position: fixed;
-  height: 100%;
-  width: 200px;
-  background-color: white;
```
Comment (rottencandy) [inline webui-src/app/theme.css:340]: Can you revert back to these changes? I'm still seeing the sidebar pushing the content panels down

File: webui-src/app/theme.css [hunk 18]
```diff
@@ -357,10 +362,12 @@ a.tab-menu-item {
   border-left: 5px solid #3ba4d7;
   animation: expand-left-border 0.1s;
 }
+.widget-2{
+  margin-top: -12rem;
+}
```
Comment (rottencandy) [inline webui-src/app/theme.css:367]: You still haven't commited the components using this class

**Review by rottencandy — COMMENTED** [review]
Thanks for the update. The chat sidebar mostly looks good, had a few other minor suggestions.

**Comment by defnax** [comment]
on google chrome the wasted space always there people, config, mail:
image

**Comment by gunjan543** [comment]
I updated your suggestions

File: .gitignore [hunk 19]
```diff
@@ -1,4 +1,3 @@
 .qmake.stash
 Makefile
-webui/*
```
Comment (Kumaravinash9) [inline .gitignore:3]: As a know about the RSwebui, the WebUI folder is the bundling of all the files of the `webui-src/app` which creates every time when we build. So, Do your changes in `webui-src /app`  folder. No need to push the `webui` folder. so please add `webui/*` in `.gitignore`.

File: webui/app.css [hunk 20]
```diff
@@ -0,0 +1,741 @@
+html, body, #main {
```
Comment (Kumaravinash9) [inline webui/app.css:1]: Remove this file

File: webui/app.js [hunk 21]
```diff
@@ -0,0 +1,5852 @@
+(function() {
```
Comment (Kumaravinash9) [inline webui/app.js:1]: same here

File: webui/index.html [hunk 22]
```diff
@@ -0,0 +1,27 @@
+<!doctype html>
+<html>
```
Comment (Kumaravinash9) [inline webui/index.html:2]: same here

Comment (Kumaravinash9) [reply webui/index.html:2]: Please remove 'webui' folder.

File: .gitignore (hunk 19 above)
Comment (gunjan543) [reply .gitignore:3]: There are changes in app.js that were not being pushed because app.js was in git ignore and without app.js, classes are not getting applied, so is there any other alternative?

Comment (Kumaravinash9) [reply .gitignore:3]: Hey @gunjan. Once look at the app.js codebase, it is exactly the same code of  `WebUI-src/app` files. So, Do your changes to that files. As I already mentioned `webui` folder is bundling of the `webui-src/app`. You can also check it in `webui-src/make-src/build.*`

Comment (gunjan543) [reply .gitignore:3]: Ok, got it ! Thankyou

**Review by rottencandy — COMMENTED** [review]
Thanks for the updates again,

just had 2 more minor code suggestions.

**Comment by Kumaravinash9** [comment]
> Thanks, just had one final comment:
> 
> * Please remove the 2 files in `webui` directory that you had commited, they are built files and should not be included in the repo.
> * It would be great if you also removed the extra ` .` that was added in the js components, doesn't seem to break anything but better to be safe.
> 
> Apart from that, the PR looks good to merge.

Hey @gunjan543 , I opened a new issue #41 in which I also mentioned the solution too. Please add these changes too in this PR. Thanks .

**Comment by gunjan543** [comment]
> > Thanks, just had one final comment:
> > 
> > * Please remove the 2 files in `webui` directory that you had commited, they are built files and should not be included in the repo.
> > * It would be great if you also removed the extra ` .` that was added in the js components, doesn't seem to break anything but better to be safe.
> > 
> > Apart from that, the PR looks good to merge.
> 
> Hey @gunjan543 , I opened a new issue #41 in which I also mentioned the solution too. Please add these changes in this PR. Thanks .

Ok I add that :)

**Comment by gunjan543** [comment]
> Thanks, just had one final comment:
> 
> * Please remove the 2 files in `webui` directory that you had commited, they are built files and should not be included in the repo.
> * It would be great if you also removed the extra ` .` that was added in the js components, doesn't seem to break anything but better to be safe.
> 
> Apart from that, the PR looks good to merge.

Sorry, but I unable to locate the extra dot that you are talking about.

File: webui-src/app/config/config_files.js [hunk 23]
```diff
@@ -9,7 +9,7 @@ const SharedDirectories = () => {
       rs.rsJsonApiRequest('/rsFiles/getSharedDirectories', {}, (data) => (directories = data.dirs));
     },
     view: (vnode) =>
-      m('.widget', [
+      m('.widget .', [
```
Comment (rottencandy) [inline webui-src/app/config/config_files.js:12]: I was mentioning these changes.

File: webui-src/app/mail/mail_resolver.js [hunk 24]
```diff
@@ -63,7 +63,7 @@ module.exports = {
     return m(
       Layout,
       m(sections[tab], {
-        list: Messages[tab],
+        list: Messages[tab].reverse(),
```
Comment (Kumaravinash9) [inline webui-src/app/mail/mail_resolver.js:66]: Thanks, @gunjan543.

**Review by rottencandy — APPROVED** [review]
Looks good. Thanks!

**Comment by gunjan543** [comment]
> Looks good. Thanks!

Thankyou
